src/test/java/org/sagebionetworks/web/unitclient/widget/table/modal/fileview/ScopeWidgetTest.java

```diff
@@ -155,41 +150,54 @@ public void testConfigureEntityViewHappyCase() {
 
-    // edit
     widget.onEditScopeAndMask();
-    verify(mockEditScopeWidget)
-      .configure(mockReferencesWithoutVersions, true, TableType.file_view);
-    verify(mockView).showModal();
-
-    // update file view to file+table view
-    when(mockView.isFileSelected()).thenReturn(true);
-    when(mockView.isTableSelected()).thenReturn(true);
-    widget.updateViewTypeMask();
-
-    // save new scope
-    widget.onSave();
-
-    verify(mockSynapseAlert).clear();
-    verify(mockView).setLoading(true);
-    // verify view type has been updated
-    // clears out old ViewType, replaced with mask
-    verify(mockEntityView).setType(null);
-    verify(mockEntityView)
-      .setViewTypeMask(
-        ViewTypeMask.getMaskForDepricatedType(ViewType.file_and_table)
-      );
-    verify(mockJsClient)
-      .updateEntity(
-        any(Table.class),
-        anyString(),
-        anyBoolean(),
-        any(AsyncCallback.class)
+
+    ArgumentCaptor<EntityViewScopeEditorModalProps.Callback> onUpdateArgumentCaptor =
+      ArgumentCaptor.forClass(EntityViewScopeEditorModalProps.Callback.class);
+    ArgumentCaptor<EntityViewScopeEditorModalProps.Callback> onCancelArgumentCaptor =
+      ArgumentCaptor.forClass(EntityViewScopeEditorModalProps.Callback.class);
+    verify(mockEntityViewScopeEditorModalWidget)
+      .configure(
+        eq(entityId),
+        onUpdateArgumentCaptor.capture(),
+        onCancelArgumentCaptor.capture(),
+        eq(true)
       );
-    verify(mockView).setLoading(false);
-    verify(mockView).hideModal();
+    onUpdateArgumentCaptor.getValue().run();
+    verify(mockEntityViewScopeEditorModalWidget).setOpen(false);
     verify(mockEventBus).fireEvent(any(EntityUpdatedEvent.class));
   }
 
+  @Test
+  public void testOnEditScopeAndMaskOnCancel() {
+    // configure with an entityview, edit the scope, and save.
+    boolean isEditable = true;
+    widget.configure(mockBundle, isEditable);
+
+    // The view scope widget does not allow edit of the scope. That occurs in the modal (with the
+    // editScopeWidget)
+    verify(mockViewScopeWidget)
+      .configure(mockReferencesWithoutVersions, false, TableType.file_view);
+    verify(mockView).setEditMaskAndScopeButtonVisible(true);
+    verify(mockView).setVisible(true);
+
+    widget.onEditScopeAndMask();
+
+    ArgumentCaptor<EntityViewScopeEditorModalProps.Callback> onUpdateArgumentCaptor =
+      ArgumentCaptor.forClass(EntityViewScopeEditorModalProps.Callback.class);
+    ArgumentCaptor<EntityViewScopeEditorModalProps.Callback> onCancelArgumentCaptor =
+      ArgumentCaptor.forClass(EntityViewScopeEditorModalProps.Callback.class);
+    verify(mockEntityViewScopeEditorModalWidget)
+      .configure(
+        eq(entityId),
+        onUpdateArgumentCaptor.capture(),
+        onCancelArgumentCaptor.capture(),
+        eq(true)
+      );
+    onCancelArgumentCaptor.getValue().run();
+    verify(mockEntityViewScopeEditorModalWidget).setOpen(false);
+  }
```

Comment:
This would pass if you had a bug where onCancel ended up calling the `onUpdate` prop. Just to make sure the correct callback was called, you should make sure `fireEvent` was never called.
```suggestion
    verify(mockEventBus, never()).fireEvent(any(EntityUpdatedEvent.class));
  }
```